velox/connectors/hive/storage_adapters/abfs/AbfsWriteFile.cpp

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/connectors/hive/storage_adapters/abfs/AbfsWriteFile.h"
+
+#include <azure/storage/files/datalake.hpp>
+
+namespace facebook::velox::filesystems::abfs {
+class BlobStorageFileClient final : public IBlobStorageFileClient {
+ public:
+  BlobStorageFileClient(const DataLakeFileClient client)
```

Comment:
`BlobStorageFileClient(const std::unique_ptr<DataLakeFileClient> client)`

Reply:
hey @majetideepak using const unique_ptr will causing can't assign to `_client` via `_client(client)`, i update to use shared_ptr, hope that also acceptable.

Reply:
@gaoyangxiaozhu You have use `std::move()` for assigning a unique pointer. Let's stick with `unique_ptr`. Thanks.

Reply:
hello @majetideepak but when i change to below

```
  BlobStorageFileClient(const std::unique_ptr<DataLakeFileClient> client)
      : client_(std::move(client)) {}
```

it shows error

```
function "std::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp> &) [with _Tp=Azure::Storage::Files::DataLake::DataLakeFileClient, _Dp=std::default_delete<Azure::Storage::Files::DataLake::DataLakeFileClient>]" (declared at line 414 of "/usr/include/c++/9/bits/unique_ptr.h") cannot be referenced -- it is a deleted function
```

Reply:
I eventually remove `const` it pass the build. 
But I don't know what's the reason here